utils/index.js

```diff
@@ -82,54 +82,64 @@ const bufferToJSON = (content, filepath) => {
  *
  * @param {String} repoName - The full repo name (username/repository)
  * @param {Array<String>} files - A list of files
- * @return {Array<String>} - The files downloaded
+ * @param {Array<String>} dependencies - A list of zel dependencies
+ * @return {Object} - An object with a list of files and dependencies that were downloaded
  */
-const fetchFiles = (repoName, files) => {
-    if (!files) {
-        throw new Error('No files found.');
+const fetchAndWrite = (repoName, { files, dependencies }) => {
+    if (!files && !dependencies) {
+        throw new Error('No files or dependencies found.');
     }
 
-    files.forEach((file) => {
-        const downloadUrl = `https://raw.githubusercontent.com/${repoName}/master/${file}`;
+    if (files && files.length) {
+        files.forEach((file) => {
+            const downloadUrl = `https://raw.githubusercontent.com/${repoName}/master/${file}`;
 
-        const fileStream = fs.createWriteStream(file);
-        https.get(downloadUrl, (resp) => {
-            resp.pipe(fileStream);
+            const fileStream = fs.createWriteStream(file);
+            https.get(downloadUrl, (resp) => {
+                resp.pipe(fileStream);
+            });
         });
-    });
+    }
 
-    return files;
+    if (dependencies && dependencies.length) {
+        dependencies.forEach((dep) => {
+            downloadRepo(dep);
+        });
+    }
+
+    return { files, dependencies };
```

Comment:
As per my first comment, we'd want to flatten a single array of `files` here only, which then gets sent to `logger`.